File: carshop-android/app/src/main/java/com/carshop/android/data/dto/Order.kt
package com.carshop.android.data.dto

import com.squareup.moshi.Json

data class Order(
    // SPEC §13:订单号是 O+18位字符串,不暴露自增 ID
    val id: String,
    @Json(name = "device_id") val deviceId: String,
    // "pending" | "paid"
    val status: String,
    @Json(name = "total_amount") val totalAmount: Int,
    @Json(name = "shipping_info") val shippingInfo: ShippingInfo,
    val items: List<OrderItem>,
    @Json(name = "created_at") val createdAt: String,
    @Json(name = "paid_at") val paidAt: String?,
)

data class ShippingInfo(
    val name: String,
    val phone: String,
    val address: String,
)

data class OrderItem(
    val id: Int,
    // SPEC §5:商品可能被删,product_id 允许 null,product_snapshot 保留快照
    @Json(name = "product_id") val productId: Int?,
    @Json(name = "product_snapshot") val productSnapshot: ProductSnapshot,
    val quantity: Int,
    val price: Int,
)

data class ProductSnapshot(
    val id: Int,
    val title: String,
    @Json(name = "product_type") val productType: String,
    val price: Int,
    val spec: String?,
    @Json(name = "main_image_url") val mainImageUrl: String,
)

// POST /api/v1/orders body
data class CreateOrderRequest(
    @Json(name = "product_id") val productId: Int,
    val quantity: Int,
)

File: carshop-admin/src/api/order.ts
import { apiClient, unwrap, type Page } from './client';

export type OrderStatus = 'pending' | 'paid';

export interface OrderItem {
  id: number;
  product_id: number | null;
  product_snapshot: {
    id: number;
    title: string;
    product_type: 'physical' | 'service_voucher';
    price: number;
    spec: string | null;
    main_image_url: string;
  };
  quantity: number;
  price: number;
}

export interface ShippingInfo {
  name: string;
  phone: string;
  address: string;
}

export interface Order {
  id: string;
  device_id: string;
  device_id_short: string; // 后台特有,后 6 位
  status: OrderStatus;
  total_amount: number; // 分
  shipping_info: ShippingInfo;
  items: OrderItem[];
  created_at: string;
  paid_at: string | null;
}

export interface OrderListParams {
  status?: OrderStatus;
  page?: number;
  page_size?: number;
}

export const orderApi = {
  list: (params: OrderListParams = {}) =>
    unwrap<Page<Order>>(apiClient.get('/admin/orders', { params })),
  detail: (id: string) => unwrap<Order>(apiClient.get(`/admin/orders/${id}`)),
};
